geopaparazzilibrary/src/eu/geopaparazzi/library/gps/GpsAvgMeasurements.java

```diff
@@ -0,0 +1,232 @@
+/*
+   Copyright 2010 Libor Tvrdik, David "Destil" Vavra
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+   Modified and added to Geopaparazzi by Tim Howard, 2015
+ */
+package eu.geopaparazzi.library.gps;
+
+import android.location.Location;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Group all the measurements. It allows the calculation of the weighted average
+ * of the measured values and record times of measurement.
+ * 
+ * @author Libor Tvrdik (libor.tvrdik@gmail.com), Destil
+ */
+public class GpsAvgMeasurements {
+
+	private static volatile GpsAvgMeasurements instance = null;
+	private final List<Location> locations;
+	private double averageLat;
+	private double averageLon;
+	private double averageAlt;
+	private float averageAccuracy;
+	private double weightedLatSum;
+	private double weightedLonSum;
+	private double altSum;
+	private double invertedAccuracySum;
+	private float distanceFromAverageCoordsSum;
+
+	/** Creates a new list. */
+	private GpsAvgMeasurements() {
+		this.locations = new ArrayList<Location>();
+		clean();
```

Comment:
I brought over this class from here:
https://github.com/tghoward/GPS-Averaging/blob/master/src/org/destil/gpsaveraging/Measurements.java

Except for the few name changes, I didn't change the class at all. The call to clean() is in the original. So I don't know if it is needed.